lib/src/rules/void_checks.dart

```diff
@@ -81,7 +81,9 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitCompilationUnit(CompilationUnit node) {
-    _futureDynamicType = context.typeProvider.futureDynamicType;
+    _futureDynamicType = context.typeSystem.leastUpperBound(
+        context.typeProvider.futureDynamicType,
+        context.typeProvider.nullType) as InterfaceType;
```

Comment:
Ah, if you need this only for `FutureOr`, instead use `typeProvider.futureOrElement.instantiate` and provide `nullabilitySuffix`. `LUB` is more expensive.